src/classes/UTIL_DMLService.cls

```diff
@@ -0,0 +1,306 @@
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT 
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS 
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE 
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, 
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, 
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; 
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER 
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT 
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN 
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE 
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+
+/**
+* @author Salesforce.org
+* @date 2017
+* @group Utilities
+* @description Service Class to handle DML actions.
+*/
+
+public with sharing class UTIL_DMLService {
+    public static final Boolean ALL_OR_NONE_FALSE = false;
+    
+
+    /*********************************************************************************************************
+    * @description Wraps the Database class insert method and defaults the DMLOptions optAllOrNone to false 
+    *   and DuplicateRuleHeader.AllowSave to true.
+    * @param recordToInsert The sObject to add to the org.
+    * @return Database.SaveResult
+    **********************************************************************************************************/
+    public static Database.SaveResult insertRecord(sObject recordToInsert) {
+        Database.DMLOptions dmlOptions = new Database.DMLOptions();
+        
+        dmlOptions.optAllOrNone = ALL_OR_NONE_FALSE;
+        dmlOptions.DuplicateRuleHeader.AllowSave = retrieveRespectDuplicateRuleSettings();
+
+        return Database.insert(recordToInsert, dmlOptions);
+    }
+
+
+    /*********************************************************************************************************
+    * @description Wraps the Database class insert method and defaults the DMLOptions optAllOrNone to false 
+    *   and DuplicateRuleHeader.AllowSave to true.
+    * @param recordsToInsert The sObjects to add to the org.
+    * @return List<Database.SaveResult>
+    **********************************************************************************************************/
+    public static List<Database.SaveResult> insertRecords(List<sObject> recordsToInsert) {
+        Database.DMLOptions dmlOptions = new Database.DMLOptions();
+        
+        dmlOptions.optAllOrNone = ALL_OR_NONE_FALSE;
```

Comment:
shouldn't this default to true since that's the behavior of doing a normal insert listSObject call?

Reply:
Same as above.